glue_vispy_viewers/scatter/scatter_toolbar.py

```diff
@@ -0,0 +1,99 @@
+__author__ = 'penny'
+
+"""
+This is for getting the selection part and highlight it
+"""
+from ..common.toolbar import VispyDataViewerToolbar
+from .layer_artist import ScatterLayerArtist
+from glue.core.edit_subset_mode import EditSubsetMode
+from glue.core.subset import ElementSubsetState
+import numpy as np
+from matplotlib import path
+
+from glue.core import Data
+
+
+class ScatterSelectionToolbar(VispyDataViewerToolbar):
+
+    def __init__(self, vispy_widget=None, parent=None):
+        super(ScatterSelectionToolbar, self).__init__(vispy_widget=vispy_widget, parent=parent)
+        self.scatter_data = None
+
+    def get_visible_data(self):
+        visible = []
+        # Loop over visible layer artists
+        for layer_artist in self._vispy_data_viewer._layer_artist_container:
+            # Only extract Data objects, not subsets
+            if isinstance(layer_artist.layer, Data):
+                visible.append(layer_artist.layer)
+        visual = layer_artist.visual  # we only have one visual for each canvas
+        return visible, visual
+
+    # TODO: implement advanced point selection here
+    def on_mouse_press(self, event):
+        self.selection_origin = event.pos
+        if self.mode is 'point':
+            # Ray intersection on the CPU to highlight the selected point(s)
+
+            data = self.tr.map(self.data)[:, :2]
+            print('data', data)
+            m1 = data > (event.pos - 4)
+            m2 = data < (event.pos + 4)
+
+            self.selected = np.argwhere(m1[:,0] & m1[:,1] & m2[:,0] & m2[:,1])
+            self.mark_selected()
```

Comment:
This won't work anymore, correct? You might have to copy some of the subset state code below here